# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/windows/sdn/sdnAccessible-inl.h
# Commit: 1a948c0ec982
# Full Hash: 1a948c0ec9826d73bba860699f540cbb25f34e34
# Author: James Teh <jteh@mozilla.com>
# Date: 2021-04-29 21:42:31
# Regressor Bug: 1694865
# File Overlap Count: 1
# Description:
#   Bug 1694865 part 18: Make EnumVariant, GeckoCustom and sdn*Accessible classes aggregate MsaaAccessible/MsaaDocAccessible instead of AccessibleWrap/DocAccessibleWrap. r=morgan
#   
#   These objects need to aggregate the primary COM object (MsaaAccessible).
#   Once the IUnknown implementation moves out of AccessibleWrap, it won't be possible to aggregate *AccessibleWrap any more.
#   
# ==============================================================================

diff -r 3ea7b46bb4f8 -r 1a948c0ec982 accessible/windows/sdn/sdnAccessible-inl.h
--- a/accessible/windows/sdn/sdnAccessible-inl.h	Thu Apr 29 10:20:56 2021 +0000
+++ b/accessible/windows/sdn/sdnAccessible-inl.h	Thu Apr 29 10:20:57 2021 +0000
@@ -19,9 +19,9 @@
   return GetExistingDocAccessible(mNode->OwnerDoc());
 }
 
-inline AccessibleWrap* sdnAccessible::GetAccessible() {
-  if (mWrap) {
-    return mWrap;
+inline MsaaAccessible* sdnAccessible::GetMsaa() {
+  if (mMsaa) {
+    return mMsaa;
   }
 
   DocAccessible* document = GetDocument();
@@ -31,9 +31,10 @@
 
   // Once we have an accessible, we should hold a reference to it so that we
   // may preserve object identity.
-  mWrap = static_cast<AccessibleWrap*>(
+  AccessibleWrap* wrap = static_cast<AccessibleWrap*>(
       document->GetAccessibleEvenIfNotInMap(mNode));
-  return mWrap;
+  wrap->GetNativeInterface(getter_AddRefs(mMsaa));
+  return mMsaa;
 }
 
 }  // namespace a11y